test/test_jsx.re

```diff
@@ -0,0 +1,567 @@
+/** Ppx Tests
+
+    This file is here to torture the jsx ppx. Tests that are directly related to
+    html or svg should go to the other files.
+*/
+open Tyxml;
+
+module type LANGUAGE = {
+  include Xml_sigs.Typed_pp;
+  type wrap('a);
+  type list_wrap('a);
+  let pp_wrap:
+    ((Format.formatter, 'a) => unit, Format.formatter, wrap('a)) => unit;
+  let pp_wrap_list:
+    ((Format.formatter, 'a) => unit, Format.formatter, list_wrap('a)) => unit;
+  let totl: list_wrap(Xml.elt) => list_wrap(elt('a));
+  let toeltl: list_wrap(elt('a)) => list_wrap(Xml.elt);
+};
+
+module TyTests = (Language: LANGUAGE) => {
+  module Testable = {
+    type t = Language.list_wrap(Xml.elt);
+    let pp = (fmt, x) =>
+      Language.pp_wrap_list(Language.pp_elt(), fmt, Language.totl(x));
+    let equal = (==);
+  };
+
+  let make = l => {
+    let f = ((name, ty1, ty2)) => (
+      name,
+      `Quick,
+      () =>
+        Alcotest.(check((module Testable)))(
+          name,
+          Language.toeltl(ty1),
+          Language.toeltl(ty2),
+        ),
+    );
+
+    List.map(f, l);
+  };
+};
+
+module Html = {
+  include Tyxml.Html;
+  let pp_wrap = pp => pp;
+  let pp_wrap_list = pp => Format.pp_print_list(~pp_sep=(_, ()) => (), pp);
+};
+module Svg = {
+  include Tyxml.Svg;
+  let pp_wrap = pp => pp;
+  let pp_wrap_list = pp => Format.pp_print_list(~pp_sep=(_, ()) => (), pp);
+};
+module HtmlTests = TyTests(Html);
+module SvgTests = TyTests(Svg);
```

Comment:
probably all this setup could be shared with the ppx tests. But I don't know if it worth the trouble

Reply:
Well, it's copy-pasted from the ppx tests. @ulrikstrid Is there a reason not to import from the ppx tests and/or factor in a test library ?